server/processors-alerts-publisher/src/main/java/org/hawkular/apm/processor/alerts/InvocationAlertsPublisherMDB.java

```diff
@@ -53,13 +55,41 @@
 
     @Override
     public void onMessage(Message message) {
-        logger.fragmentCompletionTimeReceived();
+        logger.invocationDetailsReceived();
         try {
             String data = ((TextMessage) message).getText();
-            List<CompletionTime> items = mapper.readValue(data, new TypeReference<List<CompletionTime>>() {});
-            publisher.publish(items, "FragmentCompletionTime");
+            List<NodeDetails> items = mapper.readValue(data, new TypeReference<List<NodeDetails>>() {});
+            List<Event> events = items.stream().filter(nd -> nd.isInitial())
```

Comment:
Ok, got it. It might be worth documenting this idea somewhere, as a comment perhaps. Just by looking at the code, I would take some time to come to this scenario you mentioned.

About (1) or (2), I'd stay with the simplest possible solution for now.